src/dataset/mod.rs

```diff
@@ -333,6 +335,13 @@ mod tests {
     use ndarray::{array, Array1, Array2, Axis};
     use rand::{rngs::SmallRng, SeedableRng};
 
+    #[test]
+    fn set_target_name() {
```

Comment:
Can you also add target name verification to one or two tests with `target_iter` just to verify that your iterator changes work correctly?

Reply:
addressed lines 551-556 test this.